### PR TITLE
Added Breakable Objects, Crates and Techno's Playground

### PR DESCRIPTION
**Pull request by technodragon253**

### CODE REVIEW

File: Assets/CharacterBehaviour.cs [hunk 1]
```diff
@@ -0,0 +1,75 @@
...
+
+    //MOVEMENT PROPERTIES
+    CharacterController characterController;
+    public float walkingSpeed = 5;
+    public float runningSpeed, jumpForce;
+    public bool canMove = true;
+    Vector3 moveDirection = Vector3.zero;
+
+    //PRIVATE ESSENTIALS
+    Camera playerCamera;
```
Comment (michael-ryan) [inline Assets/CharacterBehaviour.cs:19]: if this is private, make it private

File: Assets/DeleteAfterTime.cs [hunk 2]
```diff
@@ -0,0 +1,13 @@
+using System.Collections;
+using System.Collections.Generic;
+using UnityEngine;
+
+public class DeleteAfterTime : MonoBehaviour
```
Comment (michael-ryan) [inline Assets/DeleteAfterTime.cs:5]: does this need to subclass monobehaviour?

File: Assets/Scripts/Breakable.cs [hunk 3]
```diff
@@ -0,0 +1,50 @@
...
+        if (!broken.activeSelf)
+        {
+            //Set broken to be in the same position as unbroken in case the un broken object moved before breaking
+            broken.transform.position = not_broken.transform.position;
+            broken.transform.rotation = not_broken.transform.rotation;
+
+            //Enable the broken object
+            broken.SetActive(true);
+
+            //Give every boken piece the same velocity as the unbroken object in case the object was falling while being broken
```
Comment (michael-ryan) [inline Assets/Scripts/Breakable.cs:36]: typo in comment

File: Assets/Scripts/Breakable.cs [hunk 4]
```diff
@@ -0,0 +1,50 @@
...
+            //Also add an explosion force
+            Rigidbody thisRb = not_broken.GetComponent<Rigidbody>();
+            foreach (Rigidbody rb in broken.GetComponentsInChildren<Rigidbody>())
+            {
+                rb.velocity = thisRb.velocity;
+                Vector3 randomVector = new Vector3(Random.Range(0,0.5f),Random.Range(0,0.5f),Random.Range(0,0.5f));
```
Comment (michael-ryan) [inline Assets/Scripts/Breakable.cs:42]: why is this random between 0 and 0.5? why not 0 and 1?

File: Assets/Scripts/TestBreak.cs [hunk 5]
```diff
@@ -0,0 +1,18 @@
+using System.Collections;
```
Comment (michael-ryan) [inline Assets/Scripts/TestBreak.cs:1]: Should this file be in the repo? Doesn’t look like production code

**Review by michael-ryan — CHANGES_REQUESTED** [review]
Mostly LGTM. got a few notes, get back to me as this may just be my inexperience in unity talking.

**Review by michael-ryan — APPROVED** [review]
oops i meant to approve this. pls still look over the comments i made

File: Assets/DeleteAfterTime.cs (hunk 2 above)
Comment (technodragon253) [reply Assets/DeleteAfterTime.cs:5]: Yes

**Pull request merged**